4985_CommAudio/server.cpp

```diff
@@ -3,9 +3,194 @@
+        audioBuffer->close();
+    }
+}
+
+/*--------------------------------------------------------------------------------------
+--  INTERFACE:     void pauseAudio()
+--
+--  RETURNS:       void
+--
+--  DATE:          April 4, 2017
+--
+--  DESIGNER:      Alex Zielinski
+--
+--  PROGRAMMER:    Alex Zielinski
+--
+--  NOTES:
+--      Suspend audio
+---------------------------------------------------------------------------------------*/
+void pauseAudio()
+{
+    // check if audio is playing
+    if (output->state() == QAudio::ActiveState)
+    {
+        output->suspend(); // pause the audio
+    }
+}
+
+/*--------------------------------------------------------------------------------------
+--  INTERFACE:     void playAudio(QString &filePath)
+--                      QString &filePath: the filepath of audio file
+--
+--  RETURNS:       void
+--
+--  DATE:          April 4, 2017
+--
+--  DESIGNER:      Alex Zielinski
+--
+--  PROGRAMMER:    Alex Zielinski
+--
+--  NOTES:
+--      initializes audio format and audio output
+---------------------------------------------------------------------------------------*/
+void playAudio(QString &filePath)
+{
+    // check if audio was paused
+    if(output->state() == QAudio::SuspendedState)
+    {
+        output->resume();
+        //qDebug() << audioBuffer->size();
+    }
+    else
+    {
+        // create file handle for audio file
+        QFile audioFile(filePath);
+
+        // open audio file
+        if (audioFile.open(QIODevice::ReadOnly))
+        {
+            // check if audio is playing
+            if (output->state() == QAudio::ActiveState)
+            {   // check if user selected the same track
+                if (audioFilePath == filePath)
+                {
+                    //output->stop(); // stop the audio
+                    //audioBuffer->close();
+                    return;
+                }
+                else // play different track
+                {
+                    audioFilePath = filePath;
+
+                    // seek to raw audio data of wav file
+                    audioFile.seek(44);
+
+                    // extract raw audio data
+                    QByteArray audio = audioFile.readAll();
+
+                    // initialize audio buffer
+                    audioBuffer = new QBuffer(&audio);
+                    audioBuffer->open(QIODevice::ReadWrite);
+                    audioBuffer->seek(0);
+                    //qDebug() << audioBuffer->size();
+
+                    output->start(audioBuffer); // play track
+
+                    // event loop for tracck
+                    QEventLoop loop;
+                    QObject::connect(output, SIGNAL(stateChanged(QAudio::State)), &loop, SLOT(quit()));
+                    do
+                    {
+                        loop.exec();
+                    } while(output->state() == QAudio::ActiveState);
+                }
+            }
+            else
+            {
```

Comment:
I'm a little confused here:  it seems like this else statement contains the exact same code as found above, why is that?